src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Forwarders.cs

```diff
@@ -2,14 +2,33 @@
 
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.LanguageServices;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
     internal sealed partial class SyntaxTreeIndex
     {
         public ImmutableArray<DeclaredSymbolInfo> DeclaredSymbolInfos => _declarationInfo.DeclaredSymbolInfos;
 
+        public ImmutableDictionary<string, ImmutableArray<int>> SimpleExtensionMethodInfo
+            => _extensionMethodInfo.SimpleExtensionMethodInfo;
+
+        public ImmutableArray<int> ComplexExtensionMethodInfo
+            => _extensionMethodInfo.ComplexExtensionMethodInfo;
+
+        public bool ContainsExtensionMethod => SimpleExtensionMethodInfo.Count > 0 || ComplexExtensionMethodInfo.Length > 0;
+
+        public bool TryGetDeclaredSymbolInfo(int index, out DeclaredSymbolInfo declaredSymbolInfo)
+        {
+            if (index >= DeclaredSymbolInfos.Length)
```

Comment:
this woul dbe an error in roslyn right?  should we assert/contract?

Reply:
OK, fixed